### PR TITLE
Add a GUC to allow change the size of MDCache.

### PR DESCRIPTION
**Pull request by armenatzoglou**
This commit bumps up gpos and optimizer as well.

### CODE REVIEW

File: src/backend/gpopt/utils/COptTasks.cpp [hunk 1]
```diff
@@ -931,6 +931,7 @@ COptTasks::PvOptimizeTask
 	if (!CMDCache::FInitialized())
 	{
 		CMDCache::Init();
+		CMDCache::SetCacheQuota(optimizer_mdcache_size*1024);
```
Comment (gcaragea) [inline src/backend/gpopt/utils/COptTasks.cpp]: Just a heads-up, I changed this exact if clause in this commit: https://github.com/greenplum-db/gpdb/commit/b83059bc729004601b0afa9767aee9730fce196a

We'll get a conflict, but it should be easy to resolve. I think this PR should be merged first.

**Comment by gcaragea** [comment]
LGTM
:+1:

**Comment by ivannovick** [comment]
This will need some documentation right about new guc... @mkiyama are you aware of this feature?

**Comment by vraghavan78** [comment]
@foyzur @armenatzoglou Is this an external facing feature. If not I do not think this needs to be documented.

**Comment by foyzur** [comment]
@vraghavan78 this will be an external facing GUC. The customer should be able to change the cache size. @karthijrk is also running experiments to determine the optimal default value of this GUC.

File: src/backend/gpopt/utils/COptTasks.cpp (hunk 1 above)
Comment (foyzur) [inline src/backend/gpopt/utils/COptTasks.cpp]: Typically our convention is to multiply by 1024L. Also, a space before and after asterisk would be good.

**Comment by foyzur** [comment]
LGTM

**Comment by vraghavan78** [comment]
Thanks @foyzur please note for all external facing GUCs or scenarios where a release note is needed please create a doc story.

**Comment by hlinnaka** [comment]
Ping? @armenatzoglou, this pull request has been open for a long time. What's your plan with this?

**Comment by armenatzoglou** [comment]
@hlinnaka We are performing some experiments to find a good default cache size. I will change the default size and then I will merge it to master. These experiments take more time than expected cause of an issue that appeared during the experiments.

**Comment by armenatzoglou** [comment]
A few more changes have been added. I will open a new pull request for changing the default MDCache size. This guc will be documented. 

@gcaragea @foyzur @vraghavan78 Please take a look when you get a chance.

**Comment by hlinnaka** [comment]
I assume this pull request can be closed now.

**Comment by gcaragea** [comment]
Looked at the second commit, looks good. Please merge the code.

**Comment by armenatzoglou** [comment]
Pushed, thanks for review.

**Pull request closed without merging**